max-george/starter-code/server.js

```diff
@@ -0,0 +1,217 @@
+});
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(express.static('./public'));
+
+app.get('/new', function(request, response) {
+  response.sendFile('new.html', {root: './public'});
+});
+
+app.get('/articles', function(request, response) {
+  // REVIEW: This query will join the data together from our tables and send it back to the client.
+  // DONE: Write a SQL query which joins all data from articles and authors tables on the author_id value of each
+  client.query(`SELECT *
+                FROM articles
+                INNER authors
+                ON articles.author_id = authors.author_id;`)
+  .then(function(result) {
+    response.send(result.rows);
+  })
+  .catch(function(err) {
+    console.error(err)
+  });
+});
+
+app.post('/articles', function(request, response) {
+  client.query(
+    `INSERT INTO
+     authors (author, authorUrl)
+     VALUES ($1, $2)
+     ON CONFLICT (author) DO NOTHING;`, // DONE: Write a SQL query to insert a new author, ON CONFLICT DO NOTHING
+    [request.body.author,
+      request.body.authorUrl], // DONE: Add the author and "authorUrl" as data for the SQL query
+    function(err) {
+      if (err) console.error(err)
+      queryTwo() // This is our second query, to be executed when this first query is complete.
+    }
+  )
+
+  function queryTwo() {
+    client.query(
+      `SELECT author_id
+      FROM authors
+      WHERE author=$1;` , // DONE: Write a SQL query to retrieve the author_id from the authors table for the new article
+      [request.body.author], // DONE: Add the author name as data for the SQL query
+      function(err, result) {
+        if (err) console.error(err)
+        queryThree(result.rows[0].author_id) // This is our third query, to be executed when the second is complete. We are also passing the author_id into our third query
+      }
+    )
+  }
+
+  function queryThree(author_id) {
+    client.query(
+      `INSERT INTO articles (author_id, title, category, publishedOn, body)
+      VALUES ($1, $2, $3, $4, $5);`, // DONE: Write a SQL query to insert the new article using the author_id from our previous query
+      [author_id, request.body.title, request.body.category, request.body.publishedOn, request.body.body], // DONE: Add the data from our new article, including the author_id, as data for the SQL query.
+      function(err) {
+        if (err) console.error(err);
+        response.send('insert complete');
+      }
+    );
+  }
+});
+
+app.put('/articles/:id', function(request, response) {
+  // DONE: Write a SQL query to update an author record. Remember that our articles now have
+  // an author_id property, so we can reference it from the request.body.
+  // DONE: Add the required values from the request as data for the SQL query to interpolate
+  client.query(
+    `UPDATE author SET (author, authorUrl)
+    VALUES ($1, $2)
+    WHERE author_id = (SELECT author_id, FROM articles WHERE article_id=$3;`,
```

Comment:
You are missing an end parenthesis on this query. Also, the name of the table is `authors` not `author`.